staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go

```diff
@@ -1459,6 +1466,43 @@ func (vs *VSphere) NodeDeleted(obj interface{}) {
 	}
 }
 
+// Notification handler when credentials secret is added.
+func (vs *VSphere) SecretAdded(obj interface{}) {
+	klog.V(4).Infof("secret added: %+v", obj)
+	vs.refreshNodesForSecret(obj)
+}
+
+// Notification handler when credentials secret is updated.
+func (vs *VSphere) SecretUpdated(_ interface{}, newObj interface{}) {
+	klog.V(4).Infof("secret updated: %+v", newObj)
+	vs.refreshNodesForSecret(newObj)
+}
+
+// Notification handler when credentials secret is deleted.
+func (vs *VSphere) SecretDeleted(obj interface{}) {
+	klog.V(4).Infof("secret deleted: %+v", obj)
+	vs.refreshNodesForSecret(obj)
+}
+
+func (vs *VSphere) refreshNodesForSecret(obj interface{}) {
+	secret, ok := obj.(*v1.Secret)
+	if secret == nil || !ok {
+		klog.Warningf("Unrecognized secret object %+v", obj)
+		return
+	}
+
+	if secret.Name != vs.cfg.Global.SecretName ||
```

Comment:
This check and the one above should ideally happen in the event handler before calling refreshNodesForSecret.

Reply:
Agree, just happened to see 3 occurrences, so merged it into single location.